tests/conftest.py

```diff
@@ -1,28 +1,99 @@
+import os
+import urllib
 from collections.abc import Generator
+from multiprocessing import Process
+from time import sleep
+from typing import Any
+from urllib.error import URLError
 
 import pytest
+import uvicorn
+from _pytest.fixtures import SubRequest
 from fastapi.testclient import TestClient
-from sqlmodel import Session, SQLModel, create_engine
-from sqlmodel.pool import StaticPool
+from playwright.sync_api import Page, Playwright, sync_playwright
+from sqlmodel import Session
+from tad.core.config import settings
+from tad.core.db import get_engine
 from tad.main import app
 
-# needed to make sure create_all knows about the models
-from tad.models import *  # noqa: F403
+from tests.database_test_utils import DatabaseTestUtils
+
+
+class TestSettings:
+    HTTP_SERVER_SCHEME: str = "http://"
+    HTTP_SERVER_HOST: str = "127.0.0.1"
+    HTTP_SERVER_PORT: int = 8000
+    TIME_OUT: int = 10
+
+
+def run_server() -> None:
+    uvicorn.run(app, host=TestSettings.HTTP_SERVER_HOST, port=TestSettings.HTTP_SERVER_PORT)
+
+
+def wait_for_server_ready(server: Generator[Any, Any, Any]) -> None:
+    for _ in range(TestSettings.TIME_OUT):
+        try:
+            # we use urllib instead of playwright, because we only want a simple request
+            #  not a full page with all assets
+            assert urllib.request.urlopen(server).getcode() == 200  # type: ignore # noqa
+            break
+        # todo (robbert) find out what exception to catch
+        except URLError:  # server was not ready
+            sleep(1)
+
+
+@pytest.fixture(scope="module")
+def server() -> Generator[Any, Any, Any]:
+    # todo (robbert) use a better way to get the test database in the app configuration
+    os.environ["APP_DATABASE_FILE"] = "database.sqlite3.test"
+    process = Process(target=run_server)
+    process.start()
+    server_address = (
+        TestSettings.HTTP_SERVER_SCHEME + TestSettings.HTTP_SERVER_HOST + ":" + str(TestSettings.HTTP_SERVER_PORT)
+    )
+    yield server_address
+    process.terminate()
+    del os.environ["APP_DATABASE_FILE"]
+
+
+@pytest.fixture(scope="session")
+def get_session() -> Generator[Session, Any, Any]:
+    with Session(get_engine()) as session:
+        yield session
+
+
+def pytest_configure() -> None:
+    """
+    Called after the Session object has been created and
+    before performing collection and entering the run test loop.
+    """
+    # todo (robbert) creating an in memory database does not work right, tables seem to get lost?
+    settings.APP_DATABASE_FILE = "database.sqlite3.test"  # set to none so we'll use an in memory database
```

Comment:
Why not use the in memory database?

Reply:
Two reasons. First, the in memory gave errors due to session being used which I was unable to fix. Second, the E2E test has to share the database with test to be able to validate results.